fineract-core/src/main/java/org/apache/fineract/portfolio/savings/SavingsAccountTransactionType.java

```diff
@@ -56,158 +82,115 @@ public String getCode() {
-        return this.value.equals(SavingsAccountTransactionType.ANNUAL_FEE.getValue());
+        return this == ANNUAL_FEE;
     }
 
     public boolean isPayCharge() {
-        return this.value.equals(SavingsAccountTransactionType.PAY_CHARGE.getValue());
+        return this == PAY_CHARGE;
     }
 
     public boolean isChargeTransaction() {
         return isPayCharge() || isWithdrawalFee() || isAnnualFee();
     }
 
     public boolean isWaiveCharge() {
-        return this.value.equals(SavingsAccountTransactionType.WAIVE_CHARGES.getValue());
+        return this == WAIVE_CHARGES;
     }
 
     public boolean isTransferInitiation() {
-        return this.value.equals(SavingsAccountTransactionType.INITIATE_TRANSFER.getValue());
+        return this == INITIATE_TRANSFER;
     }
 
     public boolean isTransferApproval() {
-        return this.value.equals(SavingsAccountTransactionType.APPROVE_TRANSFER.getValue());
+        return this == APPROVE_TRANSFER;
     }
 
     public boolean isTransferRejection() {
-        return this.value.equals(SavingsAccountTransactionType.REJECT_TRANSFER.getValue());
+        return this == REJECT_TRANSFER;
     }
 
     public boolean isTransferWithdrawal() {
-        return this.value.equals(SavingsAccountTransactionType.WITHDRAW_TRANSFER.getValue());
+        return this == WITHDRAW_TRANSFER;
     }
 
     public boolean isWrittenoff() {
-        return this.value.equals(SavingsAccountTransactionType.WRITTEN_OFF.getValue());
+        return this == WRITTEN_OFF;
     }
 
     public boolean isDividendPayout() {
-        return this.value.equals(SavingsAccountTransactionType.DIVIDEND_PAYOUT.getValue());
+        return this == DIVIDEND_PAYOUT;
     }
 
     public boolean isIncomeFromInterest() {
-        return this.value.equals(SavingsAccountTransactionType.OVERDRAFT_INTEREST.getValue());
+        return this == OVERDRAFT_INTEREST;
     }
 
     public boolean isEscheat() {
-        return this.value.equals(SavingsAccountTransactionType.ESCHEAT.getValue());
+        return this == ESCHEAT;
     }
 
     public boolean isAmountOnHold() {
-        return this.value.equals(SavingsAccountTransactionType.AMOUNT_HOLD.getValue());
+        return this == AMOUNT_HOLD;
     }
 
     public boolean isAmountRelease() {
-        return this.value.equals(SavingsAccountTransactionType.AMOUNT_RELEASE.getValue());
+        return this == AMOUNT_RELEASE;
+    }
+
+    public boolean isCredit() {
+        // AMOUNT_RELEASE is not credit, because the account balance is not changed
+        return isCreditEntryType() && !isAmountRelease();
     }
 
     public boolean isDebit() {
-        return isWithdrawal() || isWithdrawalFee() || isAnnualFee() || isPayCharge() || isIncomeFromInterest() || isWithHoldTax()
-                || isEscheat();
+        // AMOUNT_HOLD, ESCHEAT are not debit, because the account balance is not changed
+        return isDebitEntryType() && !isAmountOnHold() && !isEscheat();
```

Comment:
I am a little confused here. The AMOUNT_HOLD and ESCHEAT is marked as DEBIT in line 50-51, but here it got excluded